File: app/src/main/java/com/mthaler/knittings/Photo.java
package com.mthaler.knittings;

import android.graphics.Bitmap;

import java.io.File;

/**
 * The photo class represents a photo. It has an id (used in the database), a filename and an
 * optional preview that can be displayed in the knittings list
 */
public class Photo {

    private final long id;
    private final File filename;
    private final long knittingID;
    private Bitmap preview;
    private String description;

    public Photo(long id, File filename, long knittingID) {
        this.id = id;
        this.filename = filename;
        this.knittingID = knittingID;
        this.description = "";
    }

    public long getId() {
        return id;
    }

    public File getFilename() {
        return filename;
    }

    public long getKnittingID() {
        return knittingID;
    }

    public Bitmap getPreview() {
        return preview;
    }

    public void setPreview(Bitmap preview) {
        this.preview = preview;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    @Override
    public String toString() {
        return "Photo{" +
                "id=" + id +
                ", filename=" + filename +
                ", knittingID=" + knittingID +
                ", description='" + description + '\'' +
                '}';
    }
}

File: app/src/main/kotlin/com/mthaler/knittings/Photo.kt
package com.mthaler.knittings

import android.graphics.Bitmap

import java.io.ByteArrayOutputStream
import java.io.File
import java.io.Serializable

/**
 * The photo class represents a photo. It has an id (used in the database), a filename and an
 * optional preview that can be displayed in the knittings list
 */
class Photo(val id: Long, val filename: File, val knittingID: Long) : Serializable {
    var preview: Bitmap? = null
    var description: String? = null

    init {
        this.description = ""
    }

    override fun toString(): String {
        return "Photo{" +
                "id=" + id +
                ", filename=" + filename +
                ", knittingID=" + knittingID +
                ", description='" + description + '\'' +
                '}'
    }

    companion object {

        fun getBytes(preview: Bitmap?): ByteArray? {
            if (preview != null) {
                val out = ByteArrayOutputStream()
                preview.compress(Bitmap.CompressFormat.JPEG, 90, out)
                return out.toByteArray()
            } else {
                return null
            }
        }
    }
}
